FWCore/Utilities/src/ObjectWithDict.cc

```diff
@@ -74,9 +76,13 @@ namespace edm {
     }
 
     if (to.hasBase(from)) { // down cast
+#ifndef _LIBCPP_VERSION
       // use the internal dynamic casting of the compiler (e.g. libstdc++.so)
       void* address = abi::__dynamic_cast(address_, static_cast<abi::__class_type_info const*>(&from.typeInfo()), static_cast<abi::__class_type_info const*>(&to.typeInfo()), -1);
```

Comment:
Quick look.
It's implemented in `src/private_typeinfo.cpp`
```
 624 extern "C" _LIBCXXABI_FUNC_VIS void *
 625 __dynamic_cast(const void *static_ptr, const __class_type_info *static_type,
 626                const __class_type_info *dst_type,
 627                std::ptrdiff_t src2dst_offset) {
```

It's not available via headers, but the symbols exists and it's exported:
```
0000000000025a33 T ___dynamic_cast
```
This is probably a safeguard to make sure that only compiler calls it (makes sense).